playback/service/src/main/java/de/danoeh/antennapod/playback/service/internal/ExoPlayerWrapper.java

```diff
@@ -264,9 +322,25 @@ public void setDisplay(SurfaceHolder sh) {
         exoPlayer.setVideoSurfaceHolder(sh);
     }
 
-    public void setPlaybackParams(float speed, boolean skipSilence) {
+    public void setPlaybackParams(final float speed, final FeedPreferences.SkipSilence skipSilence) {
         playbackParameters = new PlaybackParameters(speed, playbackParameters.pitch);
-        exoPlayer.setSkipSilenceEnabled(skipSilence);
+        // update skip silence duration in audio pipeline
+        if (skipSilence != FeedPreferences.SkipSilence.OFF && skipSilence != this.skipSilence) {
+            this.skipSilence = skipSilence;
+            final boolean wasPlaying = isPlaying();
+            final int position = getCurrentPosition();
+            exoPlayer.release();
+            createPlayer();
+            if (mediaSource != null) {
+                prepare();
+            }
+            exoPlayer.pause();
```

Comment:
I'm still getting a loud cracking noise when switching the setting. Shouldn't the pause command come before releasing?